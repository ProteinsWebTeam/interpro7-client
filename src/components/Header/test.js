// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {
  _HamburgerBtn as HamburgerBtn,
  _SideIcons as SideIcons,
  Header,
} from '.';

const renderer = new ShallowRenderer();

describe('<HamburgerBtn />', () => {
  test('open, svg, stuck', () => {
    renderer.render(<HamburgerBtn openSideNav={() => {}} open svg stuck />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('open, svg, unstuck', () => {
    renderer.render(
      <HamburgerBtn openSideNav={() => {}} open svg stuck={false} />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('open, nosvg, stuck', () => {
    renderer.render(
      <HamburgerBtn openSideNav={() => {}} open svg={false} stuck />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('open, nosvg, unstuck', () => {
    renderer.render(
      <HamburgerBtn openSideNav={() => {}} open svg={false} stuck />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('closed, svg, stuck', () => {
    renderer.render(
      <HamburgerBtn openSideNav={() => {}} open={false} svg stuck />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('closed, svg, unstuck', () => {
    renderer.render(
      <HamburgerBtn openSideNav={() => {}} open={false} svg stuck={false} />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('closed, nosvg, stuck', () => {
    renderer.render(
      <HamburgerBtn openSideNav={() => {}} open={false} svg={false} stuck />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('closed, nosvg, unstuck', () => {
    renderer.render(
      <HamburgerBtn
        openSideNav={() => {}}
        open={false}
        svg={false}
        stuck={false}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('<SideIcons />', () => {
  test('movedAway, stuck, lowGraphics', () => {
    renderer.render(<SideIcons movedAway stuck lowGraphics />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('movedAway, stuck, no lowGraphics', () => {
    renderer.render(<SideIcons movedAway stuck lowGraphics={false} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('movedAway, unstuck, lowGraphics', () => {
    renderer.render(<SideIcons movedAway stuck={false} lowGraphics />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('movedAway, unstuck, no lowGraphics', () => {
    renderer.render(<SideIcons movedAway stuck={false} lowGraphics={false} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('not movedAway, stuck, lowGraphics', () => {
    renderer.render(<SideIcons movedAway={false} stuck lowGraphics />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('not movedAway, stuck, no lowGraphics', () => {
    renderer.render(<SideIcons movedAway={false} stuck lowGraphics={false} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('not movedAway, unstuck, lowGraphics', () => {
    renderer.render(<SideIcons movedAway={false} stuck={false} lowGraphics />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('not movedAway, unstuck, no lowGraphics', () => {
    renderer.render(
      <SideIcons movedAway={false} stuck={false} lowGraphics={false} />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('<Header />', () => {
  test('stuck, signature', () => {
    renderer.render(<Header stuck isSignature stickyMenuOffset={40} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('stuck, not signature', () => {
    renderer.render(<Header stuck isSignature={false} stickyMenuOffset={40} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('unstuck, signature', () => {
    renderer.render(<Header stuck={false} isSignature stickyMenuOffset={40} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  test('unstuck, not signature', () => {
    renderer.render(
      <Header stuck={false} isSignature={false} stickyMenuOffset={40} />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
