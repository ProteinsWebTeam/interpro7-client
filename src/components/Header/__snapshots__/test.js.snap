// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HamburgerBtn /> closed, nosvg, stuck 1`] = `
<span>
  <button
    aria-label="Show the InterPro Menu"
    className="top_level_hamburger"
    onClick={[Function]}
  >
    ☰
  </button>
</span>
`;

exports[`<HamburgerBtn /> closed, nosvg, unstuck 1`] = `
<span>
  <button
    aria-label="Show the InterPro Menu"
    className="top_level_hamburger"
    onClick={[Function]}
  >
    ☰
  </button>
</span>
`;

exports[`<HamburgerBtn /> closed, svg, stuck 1`] = `
<svg
  className="top_level_hamburger stuck"
  height="2em"
  onClick={[Function]}
  viewBox="0 0 10 10"
  width="2em"
>
  <line
    className="hamb_1"
    x1="1"
    x2="9"
    y1="2"
    y2="2"
  />
  <line
    className="hamb_2"
    x1="1"
    x2="9"
    y1="5"
    y2="5"
  />
  <line
    className="hamb_3"
    x1="1"
    x2="9"
    y1="8"
    y2="8"
  />
</svg>
`;

exports[`<HamburgerBtn /> closed, svg, unstuck 1`] = `
<svg
  className="top_level_hamburger"
  height="2em"
  onClick={[Function]}
  viewBox="0 0 10 10"
  width="2em"
>
  <line
    className="hamb_1"
    x1="1"
    x2="9"
    y1="2"
    y2="2"
  />
  <line
    className="hamb_2"
    x1="1"
    x2="9"
    y1="5"
    y2="5"
  />
  <line
    className="hamb_3"
    x1="1"
    x2="9"
    y1="8"
    y2="8"
  />
</svg>
`;

exports[`<HamburgerBtn /> open, nosvg, stuck 1`] = `
<span>
  <button
    aria-label="Show the InterPro Menu"
    className="top_level_hamburger"
    onClick={[Function]}
  >
    ☰
  </button>
</span>
`;

exports[`<HamburgerBtn /> open, nosvg, unstuck 1`] = `
<span>
  <button
    aria-label="Show the InterPro Menu"
    className="top_level_hamburger"
    onClick={[Function]}
  >
    ☰
  </button>
</span>
`;

exports[`<HamburgerBtn /> open, svg, stuck 1`] = `
<svg
  className="top_level_hamburger stuck"
  height="2em"
  onClick={[Function]}
  viewBox="0 0 10 10"
  width="2em"
>
  <line
    className="hamb_1_open"
    x1="1"
    x2="9"
    y1="2"
    y2="2"
  />
  <line
    className="hamb_2_open"
    x1="1"
    x2="9"
    y1="5"
    y2="5"
  />
  <line
    className="hamb_3_open"
    x1="1"
    x2="9"
    y1="8"
    y2="8"
  />
</svg>
`;

exports[`<HamburgerBtn /> open, svg, unstuck 1`] = `
<svg
  className="top_level_hamburger"
  height="2em"
  onClick={[Function]}
  viewBox="0 0 10 10"
  width="2em"
>
  <line
    className="hamb_1_open"
    x1="1"
    x2="9"
    y1="2"
    y2="2"
  />
  <line
    className="hamb_2_open"
    x1="1"
    x2="9"
    y1="5"
    y2="5"
  />
  <line
    className="hamb_3_open"
    x1="1"
    x2="9"
    y1="8"
    y2="8"
  />
</svg>
`;

exports[`<Header /> stuck, not signature 1`] = `
<div
  className="masthead"
  id="masthead"
  style={
    Object {
      "position": "fixed",
      "top": "-40px",
    }
  }
>
  <div
    className="masthead-inner row"
  >
    <Memo(ConnectFunction) />
    <Memo(ConnectFunction)
      stuck={true}
    />
    <ResizeObserverComponent
      element="nav"
      measurements="width"
    >
      [Function]
    </ResizeObserverComponent>
  </div>
</div>
`;

exports[`<Header /> stuck, signature 1`] = `
<div
  className="masthead sign"
  id="masthead"
  style={
    Object {
      "position": "fixed",
      "top": "-40px",
    }
  }
>
  <div
    className="masthead-inner row"
  >
    <Memo(ConnectFunction) />
    <Memo(ConnectFunction)
      stuck={true}
    />
    <ResizeObserverComponent
      element="nav"
      measurements="width"
    >
      [Function]
    </ResizeObserverComponent>
  </div>
</div>
`;

exports[`<Header /> unstuck, not signature 1`] = `
<div
  className="masthead"
  id="masthead"
  style={
    Object {
      "top": "-40px",
    }
  }
>
  <div
    className="masthead-inner row"
  >
    <Memo(ConnectFunction) />
    <Memo(ConnectFunction)
      stuck={false}
    />
    <ResizeObserverComponent
      element="nav"
      measurements="width"
    >
      [Function]
    </ResizeObserverComponent>
  </div>
</div>
`;

exports[`<Header /> unstuck, signature 1`] = `
<div
  className="masthead sign"
  id="masthead"
  style={
    Object {
      "top": "-40px",
    }
  }
>
  <div
    className="masthead-inner row"
  >
    <Memo(ConnectFunction) />
    <Memo(ConnectFunction)
      stuck={false}
    />
    <ResizeObserverComponent
      element="nav"
      measurements="width"
    >
      [Function]
    </ResizeObserverComponent>
  </div>
</div>
`;

exports[`<SideIcons /> movedAway, stuck, lowGraphics 1`] = `
<div
  className="columns small-6 medium-4 lowGraphics"
>
  <div
    className="side-icons movedAway"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={true}
      svg={true}
    />
    <label
      className="side-search stuck"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;

exports[`<SideIcons /> movedAway, stuck, no lowGraphics 1`] = `
<div
  className="columns small-6 medium-4"
>
  <div
    className="side-icons movedAway"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={true}
      svg={true}
    />
    <label
      className="side-search stuck"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;

exports[`<SideIcons /> movedAway, unstuck, lowGraphics 1`] = `
<div
  className="columns small-6 medium-4 lowGraphics"
>
  <div
    className="side-icons movedAway"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={false}
      svg={true}
    />
    <label
      className="side-search"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;

exports[`<SideIcons /> movedAway, unstuck, no lowGraphics 1`] = `
<div
  className="columns small-6 medium-4"
>
  <div
    className="side-icons movedAway"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={false}
      svg={true}
    />
    <label
      className="side-search"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;

exports[`<SideIcons /> not movedAway, stuck, lowGraphics 1`] = `
<div
  className="columns small-6 medium-4 lowGraphics"
>
  <div
    className="side-icons"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={true}
      svg={true}
    />
    <label
      className="side-search stuck"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;

exports[`<SideIcons /> not movedAway, stuck, no lowGraphics 1`] = `
<div
  className="columns small-6 medium-4"
>
  <div
    className="side-icons"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={true}
      svg={true}
    />
    <label
      className="side-search stuck"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;

exports[`<SideIcons /> not movedAway, unstuck, lowGraphics 1`] = `
<div
  className="columns small-6 medium-4 lowGraphics"
>
  <div
    className="side-icons"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={false}
      svg={true}
    />
    <label
      className="side-search"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;

exports[`<SideIcons /> not movedAway, unstuck, no lowGraphics 1`] = `
<div
  className="columns small-6 medium-4"
>
  <div
    className="side-icons"
  >
    <Memo(ConnectFunction)
      aria-label="Show the InterPro Menu"
      stuck={false}
      svg={true}
    />
    <label
      className="side-search"
    >
      <Memo(ConnectFunction)
        name="search"
      />
      <Memo(ConnectFunction)
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
              "search": Object {
                "type": "text",
              },
            },
          }
        }
      >
        <div
          aria-label="Search InterPro"
        >
          <svg
            className="icon icon--search"
            height="2.2em"
            viewBox="0 0 480 480"
            width="2.2em"
          >
            <path
              d="M0,10 m250,250 a110,110 0 1,0-1,0 l0,140"
              fill="none"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="50"
              transform="rotate(-45, 328, 222)"
            />
          </svg>
        </div>
      </Memo(ConnectFunction)>
    </label>
  </div>
</div>
`;
